bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/PowerMaxBinding.java

```diff
@@ -0,0 +1,881 @@
+        }
+
+        if (serialPort == null && ipAddress == null) {
+            logger.info("PowerMax alarm binding: one connection type (Serial Port or Ip Address) must be defined");
+            return;
+        }
+
+        if (serialPort != null && ipAddress != null) {
+            logger.info(
+                    "PowerMax alarm binding: can only configure one connection type at a time: Serial Port or Ip Address");
+            return;
+        }
+
+        if (ipAddress != null) {
+            if (StringUtils.isNotBlank((String) config.get("tcpPort"))) {
+                try {
+                    tcpPort = Integer.parseInt((String) config.get("tcpPort"));
+                } catch (NumberFormatException numberFormatException) {
+                    tcpPort = 0;
+                    logger.info("PowerMax alarm binding: TCP port not configured correctly (number expected)");
+                    return;
+                }
+            }
+        }
+
+        if (StringUtils.isNotBlank((String) config.get("motionOffDelay"))) {
+            try {
+                motionOffDelay = Integer.parseInt((String) config.get("motionOffDelay")) * 60000;
+            } catch (NumberFormatException numberFormatException) {
+                motionOffDelay = DEFAULT_MOTION_OFF_DELAY;
+                logger.info("PowerMax alarm binding: motion off delay not configured correctly (number expected)");
+            }
+        }
+
+        if (StringUtils.isNotBlank((String) config.get("allowArming"))) {
+            allowArming = Boolean.valueOf((String) config.get("allowArming"));
+        }
+        if (StringUtils.isNotBlank((String) config.get("allowDisarming"))) {
+            allowDisarming = Boolean.valueOf((String) config.get("allowDisarming"));
+        }
+        if (StringUtils.isNotBlank((String) config.get("forceStandardMode"))) {
+            forceStandardMode = Boolean.valueOf((String) config.get("forceStandardMode"));
+            PowerMaxReceiveType.POWERLINK
+                    .setHandlerClass(forceStandardMode ? PowerMaxBaseMessage.class : PowerMaxPowerlinkMessage.class);
+        }
+
+        if (StringUtils.isNotBlank((String) config.get("panelType"))) {
+            try {
+                panelType = PowerMaxPanelType.fromLabel((String) config.get("panelType"));
+            } catch (IllegalArgumentException exception) {
+                panelType = PowerMaxPanelType.POWERMAX_PRO;
+                logger.info("PowerMax alarm binding: panel type not configured correctly");
+            }
+        }
+        PowerMaxPanelSettings.initPanelSettings(panelType);
+
+        if (StringUtils.isNotBlank((String) config.get("autoSyncTime"))) {
+            autoSyncTime = Boolean.valueOf((String) config.get("autoSyncTime"));
+        }
+
+        if (StringUtils.isNotBlank((String) config.get("pinCode"))) {
+            pinCode = (String) config.get("pinCode");
+        }
+
+        this.setProperlyConfigured(true);
+    }
```

Comment:
instead of the form

1. setProperlyConfigured(false)
2. parse and early return if problem
3. setProperlyConfigured(true)

I suggest the form
```java
try {
 parse
 if problem, throw new ConfigurationException("reason why")
 ...
 setProperlyConfigured(true);
} catch (ConfigurationException ce) {
 setProperlyConfigured(false);
 throw ce;
}
```